indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -764,6 +773,27 @@ def _process(self):
         return jsonify_form(form, disabled_until_change=False, submit=_('Reject'), message=message)
 
 
+class RHRegistrationsBasePrice(RHRegistrationsActionBase):
+    """Edit the base price of the selected registrations."""
+
+    def _process(self):
+        form = RegistrationBasePriceForm(base_price=self.regform.base_price, currency=self.regform.currency,
+                                         registration_id=[reg.id for reg in self.registrations])
+        if form.validate_on_submit():
+            changes = {reg: (float(reg.base_price), float(form.base_price.data)) for reg in self.registrations}
+            log_fields = {reg: f'{reg.full_name} (#{reg.friendly_id})' for reg in self.registrations}
+            for registration in self.registrations:
```

Comment:
@ThiefMaster should we do any filtering here, such as filtering by unpaid registrations?

Reply:
Yes, it should not touch registrations in any other state.

One other nice thing could be to have a checkbox in the dialog whether to (also) apply to "complete" registrations that have a zero price at the moment (those would move to "unpaid" then)...

So the full logic would be:

- state unpaid -> update price
- checkbox not ticked AND any other state -> no action
- checkbox ticked AND state complete AND no current price -> update price

Reply:
alright, sounds good

Reply:
also, should we allow setting the fee to zero? if so, whats the procedure - change the reg state to complete?

Reply:
hmm good question... maybe it'd be better to have a dropdown / radio button list to switch what you want to do first?

- Remove fee (marks registration as complete if no fields require payment) [no input shown in that case]
- Sync fee to regform default (xx CHF) [no input shown]
- Change fee to custom value [input shown]

Reply:
done. the logic got quite a bit complicated, so I think it's worth adding some unit tests